src/main/java/com/dnastack/gatekeeper/token/TokenParser.java

```diff
@@ -4,46 +4,50 @@
 import io.jsonwebtoken.Jws;
 import io.jsonwebtoken.JwtException;
 import io.jsonwebtoken.JwtParser;
+import lombok.Setter;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.context.properties.ConfigurationProperties;
 import org.springframework.stereotype.Component;
+import org.springframework.validation.annotation.Validated;
 
+import javax.validation.constraints.NotEmpty;
 import java.util.List;
 import java.util.Optional;
 
 import static java.lang.String.format;
 
 @Slf4j
 @Component
+@Validated
+@ConfigurationProperties(prefix = "gatekeeper.token")
 public class TokenParser {
 
     @Autowired
     private JwtParser jwtParser;
-    // Can't default to empty list when we specify value in application.yml
-    @Value("${gatekeeper.token.audiences:#{T(java.util.Collections).emptyList()}}")
-    private List<String> acceptedAudiences;
+    @NotEmpty
```

Comment:
Can you configure an error message here? It would be great if we can tell the user that they need to configure an audience.

Reply:
There is following message by default, it seemed fine to me, what do you think?
***************************
APPLICATION FAILED TO START
***************************

Description:

Binding to target org.springframework.boot.context.properties.bind.BindException: Failed to bind properties under 'gatekeeper.token' to com.dnastack.gatekeeper.token.TokenParser failed:

    Property: gatekeeper.token.audiences
    Value: []
    Reason: can not be empty

Reply:
Yeah, LGTM.